src/main/scala/Engine.scala

```diff
@@ -41,8 +41,9 @@ case class Query(
     returnSelf: Option[Boolean] = None,// means for an item query should the item itself be returned, defaults
                                        // to what is in the algorithm params or false
     num: Option[Int] = None, // default: whatever is in algorithm params, which itself has a default--probably 20
-    eventNames: Option[List[String]]) // names used to ID all user actions
-  extends Serializable
+    eventNames: Option[List[String]], // names used to ID all user actions
+    addRank: Option[Boolean] = None // Add to ItemScore rank fields values
+) extends Serializable
 
```

Comment:
no this is in engine.json because the index must be created to have these fields attached to items. I suppose the precedence could be in the query or overridden in the query but since it is not likely to change ever, why force it in the query. Putting in engine.json would allow us to use or not-use any particular ranking method, allowing us to not calculate one that is not desired.